### ISSUE TITLE
problem with lstm_seq2seq with word embeddings

### ISSUE THREAD

**Issue by menajosep** [keras-team/keras#9057]
Hi all,

Thanks in advance for your help. I'm trying to implement a seq2seq using keras and following the instructions given in the ten minutes tutorial(https://blog.keras.io/a-ten-minute-introduction-to-sequence-to-sequence-learning-in-keras.html). In my case I need to used word embeddings, son I followed the isntructions given in the FAQ section. This is my resulting model:
```

# Define an input sequence and process it.
encoder_inputs = Input(shape=(None,))
enc = encoder_embeddings(encoder_inputs)
enc, state_h, state_c = LSTM(vector_dim, return_state=True)(enc)
# We discard `encoder_outputs` and only keep the states.
encoder_states = [state_h, state_c]

# Set up the decoder, using `encoder_states` as initial state.
decoder_inputs = Input(shape=(None,))
dec = decoder_embeddings(decoder_inputs)
decoder_outputs = LSTM(vector_dim, return_sequences=True)(dec, initial_state=encoder_states)
decoder_dense = Dense(decoder_embeddings.input_dim, activation='softmax', name='output')
decoder_outputs = decoder_dense(decoder_outputs)

# Define the model that will turn
# `encoder_input_data` & `decoder_input_data` into `decoder_target_data`
model = Model([encoder_inputs, decoder_inputs], decoder_outputs)
```

encoder and decoder_embeddings layer is passed as parameter and consists of a trained embedding layer of dim = 128.
Inputs are of this shape:
```
encoder_input_data = np.zeros(
    (len(abstracts), max_encoder_words),
    dtype='float32')
decoder_input_data = np.zeros(
    (len(titles), max_decoder_words),
    dtype='float32')
decoder_target_data = np.zeros(
    (len(titles), max_decoder_words, len(dictionary)),
    dtype='float32')
```
Where input_data are sequences of word indexes and target_data is one hot encoding of the words.
The things is that at training time, Keras is telling me that the output passed to fit does not match the expected shape:
expected output to have shape (None, 1, 10000) but got array with shape (1024, 17, 10000)
My vocabulary is made of 100000 words, with a batch size of 1024 and the max decoder words is 17.
I've been checking at run time my implementation and the example in keras, and the difference comes in the output of the Dense layer. In the example, the output  has a shape of (?,?,93) while in my implementation the output has the second dimension fixed to 1 (?, 1, 10000). I don't know why, after applying the dense to the LSTM output it is guessing this dimension of 1. The shape of the inputs, in my case after applying the embedding layer, are similar: (?,?,93) in the original example, (?,?,128) in my case.
Anyone idea of what is happening?

**Comment by mad-bhu** [comment]
having a similar problem. Did you get to solve it?
Thanks
